### PR TITLE
Add bandwidth module

### PR DESCRIPTION
**Pull request by joshliebe**
Task/Issue URL: https://app.asana.com/0/1200019156869587/1201894596703922/f

### Description
This PR adds the bandwidth usage worker for AppTP beta users.

### Steps to test this PR

_As an AppTP beta user_
- [ ] Verify that the worker is scheduled.

_As a non-AppTP beta user_
- [ ] Verify that the worker is not scheduled.

### CODE REVIEW

File: bandwidth/bandwidth-impl/src/main/java/com/duckduckgo/bandwidth/impl/BandwidthWorker.kt [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            worker.bandwidthCollector = bandwidthCollector
+            return true
+        }
+        return false
+    }
+}
+
+class BandwidthScheduler(
+    private val workManager: WorkManager,
+    private val atpWaitlistRepository: AtpWaitlistStateRepository
```
Comment (aitorvs) [inline bandwidth/bandwidth-impl/src/main/java/com/duckduckgo/bandwidth/impl/BandwidthWorker.kt:76]: I think you should probably use the `AtpCohortManager` that is in the `:vpn-api` module.
If the user has a non-null cohort, then it has enabled AppTP previously.

The `:vpn` module is effectively the `:vpn-impl` module, and `impl` modules should not depend on other `impl` modules :D

File: bandwidth/bandwidth-impl/src/main/java/com/duckduckgo/bandwidth/impl/BandwidthWorker.kt [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+) : WorkerInjectorPlugin {
+
+    override fun inject(worker: ListenableWorker): Boolean {
+        if (worker is BandwidthWorker) {
+            worker.bandwidthCollector = bandwidthCollector
+            return true
+        }
+        return false
+    }
+}
```
Comment (aitorvs) [inline bandwidth/bandwidth-impl/src/main/java/com/duckduckgo/bandwidth/impl/BandwidthWorker.kt:72]: Very soon we [won't need](https://app.asana.com/0/0/1202122547708149) to manually do this anymore 🤞

File: bandwidth/bandwidth-store/src/main/java/com/duckduckgo/bandwidth/store/BandwidthEntity.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package com.duckduckgo.bandwidth.store
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+
+@Entity(tableName = "bandwidth")
+data class BandwidthEntity(
+    @PrimaryKey val id: Int = 0,
```
Comment (aitorvs) [inline bandwidth/bandwidth-store/src/main/java/com/duckduckgo/bandwidth/store/BandwidthEntity.kt:24]: Am I correct the primary key will always be `0` so effectively we'll never have more than one item in the table?
If so I'd probably leave a comment right here in the entity

File: bandwidth/bandwidth-store/src/main/java/com/duckduckgo/bandwidth/store/BandwidthDao.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import androidx.room.Query
+
+@Dao
+interface BandwidthDao {
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insert(bandwidthEntity: BandwidthEntity)
+
+    @Query("SELECT * FROM bandwidth")
+    fun getBandwidthEntity(): BandwidthEntity?
```
Comment (aitorvs) [inline bandwidth/bandwidth-store/src/main/java/com/duckduckgo/bandwidth/store/BandwidthDao.kt:31]: nit: rename to `getBandwidth` (?)

**Review by aitorvs — APPROVED** [review]
LGTM.
The most important comment for me is [this one](https://github.com/duckduckgo/Android/pull/1866/files#r852033158) around using the cohort that is in the `:vpn-api` rather than the repository that is in the `:vpn` module.

Assuming that'd be corrected, approving the PR

File: bandwidth/bandwidth-impl/build.gradle [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+apply from: "$rootProject.projectDir/gradle/android-library.gradle"
+
+dependencies {
+    implementation project(path: ':bandwidth-store')
+    api project(path: ':bandwidth-store')
+    implementation project(path: ':common')
+    implementation project(path: ':di')
+    implementation project(path: ':browser-api')
+    implementation project(path: ':statistics')
+    implementation project(path: ':vpn')
```
Comment (aitorvs) [inline bandwidth/bandwidth-impl/build.gradle:32]: With the comment [here](https://github.com/duckduckgo/Android/pull/1866/files#r852033158) we probably can remove this dependency in favor of `:vpn-api` (?)

**Comment by joshliebe** [comment]
> LGTM. The most important comment for me is [this one](https://github.com/duckduckgo/Android/pull/1866/files#r852033158) around using the cohort that is in the `:vpn-api` rather than the repository that is in the `:vpn` module.
> 
> Assuming that'd be corrected, approving the PR

Thanks for the review @aitorvs, I'll address the comments 👍

**Pull request merged**